.editorconfig

```diff
@@ -1,4 +1,130 @@
 root = true
 
 [*.cs]
-dotnet_diagnostic.CA1822.severity = none
\ No newline at end of file
+dotnet_diagnostic.CA1822.severity = none
+# To learn more about .editorconfig see https://aka.ms/editorconfigdocs
+###############################
+# Core EditorConfig Options   #
+###############################
+# All files
+[*]
+indent_style = space
+# Code files
+[*.{cs,csx,vb,vbx}]
+indent_size = 4
+insert_final_newline = warning
+charset = utf-8-bom
+###############################
+# .NET Coding Conventions     #
+###############################
+[*.{cs,vb}]
+dotnet_analyzer_diagnostic.category-Style.severity = warning
```

Comment:
I did a diff between this .editorconfig and the export of the default C# rules from an up-2-date Visual Studio 2019 C# settings w/o any modifications from my side (I barely use Visual Studio  nowadays). There are many many differences.

Let us discuss this in the PR before I further review here. Where you got that file from?

Reply:
Actually I use VS Enterprise 2019 Preview 1.0 (16.10.0 Preview 1.0). My .editorconfig has been generated from Visual Studio using Add New Item --> Visual C# Items --> editorconfig File (.NET). This is/should be, according to MS, the defacto-standard for C#.

I just compared a newly generated one with the one in the PR and see folowing differences:
 - dotnet_analyzer_diagnostic.category-Style.severity = warning is not the default; I took that in for getting warnings about my "style-problems" (remember my first PR:-)); the default is "suggestion" instead of warning
 - dotnet_sort_system_directives_first = true (see also below on one of your comments) IS DEFAULT in the newly generated one
 - dotnet_separate_import_directive_groups = true is NOT in the template; I cannot remember adding that one to my .editorconfig; it adds those additional newlines between the groups of using-statments
 - dotnet_naming_rule.constant_fields_should_be_pascal_case.severity = warning defaults to suggestion also; same reason here

The accessibility modifiers (internal, private above) were suggestions made by VS 2019 to me. I looked up and this can be handled by:
 dotnet_style_require_accessibility_modifiers  but I cannot find any option for that in the options-pane :-(

csharp_style_unused_value_assignment_preference = discard_variable is really not set in both .editorconfigs, but it is set in VS 2019 ("Avoid unused value assignments : Discard : Suggestion" in Options).
I must say, that I really like this discard! 
Did you ever do something like:
````c#
Console.WriteLine("Hello World!");
var str = "Hällo";
str.Replace("ä", "ae");
Console.WriteLine($"Content of str: {str}");
````
and wondering why the output is: "Content of str: Hällo"?
Clear, not with strings, because we all know this (reasonable?) behaviour and know that strings are immutable. But what is about other situations where it is maybe not so clear? with the discard you (and also later maintainers) see clearly that you ignore the result by intention (so you've thought about it). see more on [Discards - C# Guide](https://docs.microsoft.com/en-us/dotnet/csharp/discards)
If I see it correct, you're using discards only in switch-statements, right?

I will do change on this PR #166 regarding the examlpe which just has the change in Program.cs in reagdrs to the call of example.InitHostAndDiscoverAsync().
Then we can open up a discussion issue if you like about the code-styling

Reply:
Thanks for removing it. This discards style is absolute valueable when you do resource management ... but C# as a GC does not typically do this explicitely. This code is seriously disturbing. I have to read into this.

Regards how you created the editorconfig ... i went to VS 2019, options->text editor->c#->...-> generate editorconfig.

I have to work that out. I want to learn this editorconfig concept and also understand this special case about the discards into perfection (because I want myself and my projects close to the defaults of C#). But there are differences between VS and VS Code / C# plugin etc. So like that, need some research time on that.

Reply:
> VS 2019, options->text editor->c#->...-> generate editorconfig

This actually writes out your settings(options, which may include manual changes in the options (even if you have more instances of VS running and use a MS-account they reaply these settings between the instances; some options my come from older versions (using updates),...)
Because this is so unpredictable, I actually tend to use an .editorconfig-file with root=true in the base-dir of the solution. Actually I don't check it into GitHub...
You've got also an .editorconfig in the solution-base-dir; but it just includes 
dotnet_diagnostic.CA1822.severity = none

But you're absolutly right - this is not so urgent....